### PR TITLE
rec: Fix some warnings reported by cppcheck

### PR DESCRIPTION
**Pull request by chbruyand**
### Short description
This also removes some unused code.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled and tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/arguments.cc [hunk 1]
```diff
@@ -100,7 +100,7 @@ bool ArgvMap::contains(const string &var, const string &val)
   vector<string>::const_iterator i;
   
   stringtok( parts, param->second, ", \t" );
-  for( i = parts.begin(); i != parts.end(); i++ ) {
+  for( i = parts.begin(); i != parts.end(); ++i ) {
```
Comment (rgacogne) [inline pdns/arguments.cc]: I wouldn't mind a C++11 loop instead :)

File: pdns/arguments.cc [hunk 2]
```diff
@@ -118,7 +118,7 @@ string ArgvMap::helpstring(string prefix)
   
   for(map<string,string>::const_iterator i=helpmap.begin();
       i!=helpmap.end();
```
Comment (rgacogne) [inline pdns/arguments.cc]: Same

File: pdns/arguments.cc [hunk 3]
```diff
@@ -154,7 +154,7 @@ string ArgvMap::configstring(bool current)
   else
     help="# Autogenerated configuration file template\n";
   
-  for(map<string,string>::const_iterator i=helpmap.begin(); i!=helpmap.end(); i++) {
+  for(map<string,string>::const_iterator i=helpmap.begin(); i!=helpmap.end(); ++i) {
```
Comment (rgacogne) [inline pdns/arguments.cc]: Here too

File: pdns/resolver.cc [hunk 4]
```diff
@@ -115,7 +115,7 @@ Resolver::Resolver()
 
 Resolver::~Resolver()
 {
-  for(std::map<std::string,int>::iterator iter = locals.begin(); iter != locals.end(); iter++) {
+  for(std::map<std::string,int>::iterator iter = locals.begin(); iter != locals.end(); ++iter) {
```
Comment (rgacogne) [inline pdns/resolver.cc]: C++11 loop here too?

File: pdns/arguments.cc (hunk 1 above)
Comment (chbruyand) [reply pdns/arguments.cc]: Fair enough !

**Pull request merged**